force-app/main/default/classes/symptomExtension.cls

```diff
@@ -1,80 +1,98 @@
+				Map<String, String> presenting = new Map<String, String>();
+				presenting.put('id', s.InfermedicaId__c);
+				presenting.put('choice_id', 'present');
+				presentingSymptoms.add(presenting);
+			}
+		}
+		ApexPages.Message myMsg = new ApexPages.Message(ApexPages.Severity.INFO, 'Symptoms: ' + PresentingSymptoms);
+		ApexPages.addMessage(myMsg);
+		insertPossibleConditions();
+		return goToPossibleConditions();
+	}
+	
+    public PageReference goToPossibleConditions() {
 
-    public void submit() {
-        presentingSymptoms = new List<Map<String, String>>();
-        for (Symptom__c s : ctrl) {
-            if (s.Present__c == true) {
-                Map<String, String> presenting = new Map<String, String>();
-                presenting.put('id', s.InfermedicaId__c);
-                presenting.put('choice_id', 'present');
-                presentingSymptoms.add(presenting);
-            }
-        }
-        insertPossibleConditions();
+        String url = '/apex/ConditionChecker';
+        Pagereference pageRef = new Pagereference(url);
+
+        return pageRef;
     }
 
-    public void insertPossibleConditions() {
-        APIInput a = new APIInput();
-        a.sex = userContact.Sex__c.toLowerCase();
-        a.age = String.valueOf(userContact.Age__c);
-        a.evidence = presentingSymptoms;
-        List<Object> conditions = InfermedicaTester.sendRequest(a);
 
-        List<String> conditionNames = new List<String>();
-        List<Map<String, Object>> conditionList = new List<Map<String, Object>>();
-        List<PossibleCondition__c> newPossibleConditions = new List<PossibleCondition__c>();
+	public void insertPossibleConditions() {
+		try{
+			APIInput a = new APIInput();
+			a.sex = userContact.Sex__c.toLowerCase();
+			a.age = String.valueOf(userContact.Age__c);
+			a.evidence = presentingSymptoms;
+			List<Object> conditions = InfermedicaTester.sendRequest(a);
 
-        for (Object condition : conditions) {
-            String s = JSON.serialize(condition);
-            Map<String, Object> m = (Map<String, Object>)JSON.deserializeUntyped(s);
-            conditionList.add(m);
-            conditionNames.add((String)m.get('name'));
-        }
+			List<Condition__c> newConditions = new List<Condition__c>();
+			List<Map<String, Object>> conditionList = new List<Map<String, Object>>();
+			List<PossibleCondition__c> newPossibleConditions = new List<PossibleCondition__c>();
 
-        List<Condition__c> conditionRecords = [
-            SELECT Id, Name 
-            FROM Condition__c 
-            WHERE Name 
-            IN :conditionNames
-        ];
+			for (Object condition :conditions) {
+				String s = JSON.serialize(condition);
+				Map<String, Object> m = (Map<String, Object>) JSON.deserializeUntyped(s);
+				conditionList.add(m);
+				Condition__c c = new Condition__c();
+				c.Name = (String) m.get('name');
+				c.InfermedicaId__c = (String) m.get('id');
+				newConditions.add(c);
+			} upsert newConditions;
```

Comment:
The **upsert** DML operation creates new records and updates sObject records within a single statement, using a specified field to determine the presence of existing objects, **or the ID field** if no field is specified.

The SObject Id will be different always so this command will insert the same conditions again and again to Conditions table. Update will never happen. 

I recommend these changes
1. Create a field that will contain the number portion of an infermedica id, in Symptom table 
 Field Name: InfermedicaExternalID__c
Data Type: Number, **External Id, Unique**
2. Update data creator to include values for this new field or manually edit the existing Symptom records
3.  Use this code
```
for (Object condition :conditions) {
				String s = JSON.serialize(condition);
				Map<String, Object> m = (Map<String, Object>) JSON.deserializeUntyped(s);
				conditionList.add(m);
				Condition__c c = new Condition__c();
				c.Name = (String) m.get('name');
				c.InfermedicaId__c = (String) m.get('id');
                                **c.InfermedicaExternalID__c = Integer.valueOf(c.InfermedicaId__c .split('_')[1]);**
				newConditions.add(c);
			} 
                        **upsert newConditions c.InfermedicaId__c ;**
```

Reply:
Here is a code for testing


```
Condition__c oldCondition = [SELECT Name, InfermedicaId__c, InfermedicaExternalID__c FROM Condition__c WHERE Name LIKE 'Hang%' LIMIT 1].get(0);
Condition__c newCondition = new Condition__c();
newCondition.InfermedicaExternalID__c = oldCondition.InfermedicaExternalID__c;
newCondition.InfermedicaId__c = oldCondition.InfermedicaId__c;
newCondition.Name = oldCondition.Name + 'updatedupdated';

upsert newCondition Condition__c.Fields.InfermedicaExternalID__c;

System.debug([SELECT Name FROM Condition__c]);
System.debug(Integer.valueOf('c_3911'.split('_')[1]));
```